### PR TITLE
Removing non relevant argument binding_kind of GLocalDef.

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:**  cleanup

This is an easy self-contained simplification commit taken out of #13445 so as to simplify the review of the latter.

### CODE REVIEW

File: interp/constrintern.ml [hunk 1]
```diff
@@ -572,10 +572,10 @@ let intern_assumption intern ntnvars env nal bk ty =
 
 let glob_local_binder_of_extended = DAst.with_loc_val (fun ?loc -> function
   | GLocalAssum (na,bk,t) -> (na,bk,None,t)
-  | GLocalDef (na,bk,c,Some t) -> (na,bk,Some c,t)
-  | GLocalDef (na,bk,c,None) ->
+  | GLocalDef (na,c,Some t) -> (na,Explicit,Some c,t)
```
Comment (SkySkimmer) [inline interp/constrintern.ml:575]: Is this somehow dropped?

Comment (herbelin) [reply interp/constrintern.ml:575]: `bk` was always `Explicit` for `GLocalDef`. This moves the time we set the `bk` of a local def to a place where the structure does not allow us to drop it.

Told in another way, there is not enough structure in this quadruple to indicate that the `bk` is forced when the third argument is `Some`. (Maybe a sign that we should not use this quadruple anymore??)

**Comment by herbelin** [comment]
@skyskimmer: would you like to assign?

**Comment by herbelin** [comment]
@skyskimmer: would you like to assign?

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot: merge now

**Pull request merged**